syft/pkg/package.go

```diff
@@ -27,6 +27,7 @@ type Package struct {
 	Type      Type             `cyclonedx:"type"`                   // the package type (e.g. Npm, Yarn, Python, Rpm, Deb, etc)
 	CPEs      []cpe.CPE        `hash:"ignore"`                      // all possible Common Platform Enumerators (note: this is NOT included in the definition of the ID since all fields on a CPE are derived from other fields)
 	PURL      string           `hash:"ignore"`                      // the Package URL (see https://github.com/package-url/purl-spec)
+	Digests   []file.Digest    `hash:"ignore"`                      // the digests of the files that make up this package
```

Comment:
I don't think having a slice of digests here is the right thing to do. These are not associated with files, but are supposed to be digest of files. I can see that the ArchiveDigest seems similar, but it has known semantics for Java packages that were identified in a JAR file, where this field would not make sense for other types, and we should almost always be putting digests associated with a file.